airflow/providers/mongo/hooks/mongo.py

```diff
@@ -118,6 +155,21 @@ def __init__(self, mongo_conn_id: str = default_conn_name, *args, **kwargs) -> N
             # Case: HTTP (ssl=False) with allow_insecure not specified
             self.allow_insecure = False
 
+    @staticmethod
+    def _validate_connection(conn: Connection):
+        conn_type = conn.conn_type
+        if conn_type != "mongo":
+            if conn_type == "mongodb+srv":
+                raise AirflowException(
+                    "Mongo SRV connections should have the conn_type 'mongo' and set 'use_srv=true' in extras"
+                )
+            raise AirflowException(
+                f"conn_type '{conn_type}' not allowed for MongoHook; conn_type must be 'mongo'"
+            )
+
+        if conn.port and "srv" in conn.extra_dejson and conn.extra_dejson["srv"] is True:
+            raise AirflowException("srv URI should not specify a port")
```

Comment:
Can these be more specific than AirflowException?

Reply:
Sure, I'll switch to `AirflowConfigException`.

Reply:
Done